File: Week 1/id_720/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        // 空间换时间
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i=0; i<nums.length; i++) {

            int tmp = target-nums[i];
            
            if (map.containsKey(tmp)) {
                return new int[]{map.get(tmp), i};
            } 
            
            map.put(nums[i], i);
        }
        
        throw new IllegalArgumentException("No two sum solution");
    }
}

// class Solution {
//     public int[] twoSum(int[] nums, int target) {
        
//         int[] res = new int[2];
        
//         for (int i=0; i<nums.length-1; i++) {
            
//             for (int j=i+1; j<nums.length; j++) {
                
//                 if (target == (nums[i] + nums[j])) {
//                     res[0] = i;
//                     res[1] = j;
//                 }
//             }
//         }
        
//         return res;
//     }
// }

File: Week 1/id_530/twoSum.php
<?php
class Solution {

    /**
	 * 数组任意俩元素求和
	 * @url https://leetcode.com/problems/two-sum/
     * @param Integer[] $arr
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($arr, $target) {
        for($i=0; $i<count($arr); $i++) {
			$diff = $target - $arr[$i];
			// array_search如果在数组中找到指定的键值，则返回对应的键名，否则返回 FALSE
			$index = array_search($diff, $arr);
			if($index !==false &&  $index != $i) {
				return [$i, $index];
			}
		}
		
		return [];
    }
}